File: client-ios/Sources/Data/SettingsRepository.swift
import Foundation

/// Minimal persistent settings (node URL, voice consent).
///
/// Only stores necessary configuration, no tracking data.
@MainActor
class SettingsRepository: ObservableObject {
    static let defaultNodeURL = "http://localhost:8000"
    
    @Published var nodeUrl: String {
        didSet {
            UserDefaults.standard.set(nodeUrl, forKey: "nodeUrl")
        }
    }
    
    @Published var voiceEnabled: Bool {
        didSet {
            UserDefaults.standard.set(voiceEnabled, forKey: "voiceEnabled")
        }
    }
    
    @Published var voiceConsentGiven: Bool {
        didSet {
            UserDefaults.standard.set(voiceConsentGiven, forKey: "voiceConsentGiven")
        }
    }
    
    init() {
        self.nodeUrl = UserDefaults.standard.string(forKey: "nodeUrl") ?? Self.defaultNodeURL
        self.voiceEnabled = UserDefaults.standard.bool(forKey: "voiceEnabled")
        self.voiceConsentGiven = UserDefaults.standard.bool(forKey: "voiceConsentGiven")
    }
}

File: client-android/src/main/java/com/ambient/client/data/SettingsRepository.kt
package com.ambient.client.data

import android.content.Context
import androidx.datastore.core.DataStore
import androidx.datastore.preferences.core.Preferences
import androidx.datastore.preferences.core.booleanPreferencesKey
import androidx.datastore.preferences.core.edit
import androidx.datastore.preferences.core.stringPreferencesKey
import androidx.datastore.preferences.preferencesDataStore
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.map

/**
 * Minimal persistent settings (node URL, voice consent).
 * 
 * Only stores necessary configuration, no tracking data.
 */
private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "settings")

class SettingsRepository(private val context: Context) {
    companion object {
        private val NODE_URL_KEY = stringPreferencesKey("node_url")
        private val VOICE_ENABLED_KEY = booleanPreferencesKey("voice_enabled")
        private val VOICE_CONSENT_KEY = booleanPreferencesKey("voice_consent_given")
        
        const val DEFAULT_NODE_URL = "http://localhost:8000"
    }

    val nodeUrl: Flow<String> = context.dataStore.data.map { prefs ->
        prefs[NODE_URL_KEY] ?: DEFAULT_NODE_URL
    }

    val voiceEnabled: Flow<Boolean> = context.dataStore.data.map { prefs ->
        prefs[VOICE_ENABLED_KEY] ?: false
    }

    val voiceConsentGiven: Flow<Boolean> = context.dataStore.data.map { prefs ->
        prefs[VOICE_CONSENT_KEY] ?: false
    }

    suspend fun setNodeUrl(url: String) {
        context.dataStore.edit { prefs ->
            prefs[NODE_URL_KEY] = url.trim()
        }
    }

    suspend fun setVoiceEnabled(enabled: Boolean) {
        context.dataStore.edit { prefs ->
            prefs[VOICE_ENABLED_KEY] = enabled
        }
    }

    suspend fun setVoiceConsent(given: Boolean) {
        context.dataStore.edit { prefs ->
            prefs[VOICE_CONSENT_KEY] = given
        }
    }
}
